### PR TITLE
[Android] enableReadingModeQuickMenu prop

### PR DESCRIPTION
**Pull request by darrenchann**
prop for disableReadingMode
set `enableReadingModeQuickMenu={false}` to disable quick menu during reflow mode

pending test instruction
Ensure that props are enabled
App.js

```
<DocumentView
enableReadingModeQuickMenu={false}
```

- long press your viewer to see that quick menu is disabled

### CODE REVIEW

**Review by ama-pdftron — CHANGES_REQUESTED** [review]
Please edit API.md to include this new APi, almost mention ANdroid-only
https://github.com/PDFTron/pdftron-react-native/blob/master/API.md

**Comment by ama-pdftron** [comment]
Doesn't seem to run:
image

**Comment by darrenchann** [comment]
@ama-pdftron 

provided incorrect test instructions with the prop - also updated api.md
still need to repair local dev environment/testing environment 😵‍💫

File: API.md [hunk 1]
```diff
@@ -3971,4 +3971,15 @@ fieldHightlightColor | object | the form field highlight color, in the format `{
 
 ```js
 this._viewer.setFormFieldHighlightColor({red: 0, green: 0, blue: 255, alpha: 128}); // blue color
+```
+
+#### disableReadingModeQuickMenu
+bool, optional, defaults to false
+
+Defines whether the viewer disables long press quick menu in read-only. If true, the UI will not allow the user to long press to open the quick menu.
```
Comment (ama-pdftron) [inline API.md:3979]: Mention `Android Only`

**Review by ama-pdftron — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java [hunk 2]
```diff
@@ -770,6 +770,10 @@ public void setCurrentToolbar(String toolbarTag) {
     }
 
     // Hygen Generated Props
+    public void setDisableReadingModeQuickMenu(boolean disabled) {
+        getToolManager().setReflowTextSelectionMenuEnabled(disabled);
```
Comment (ama-pdftron) [inline android/src/main/java/com/pdftron/reactnative/views/DocumentView.java]: You need to inverse the logic
The flag is to disable the control

`setReflowTextSelectionMenuEnabled(true)` means the menu is ENABLED
`setReflowTextSelectionMenuEnabled(false)` means the menu is DISABLED

**Review by ama-pdftron — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java (hunk 2 above)
Comment (darrenchann) [reply android/src/main/java/com/pdftron/reactnative/views/DocumentView.java]: i inversed the logic and renamed the prop setEnabledReadingModeQuickMenu - makes it more consistent to the native api that was added on android

**Review by ama-pdftron — CHANGES_REQUESTED** [review]
Do not check in `yarn.lock`

File: example/App.js [hunk 3]
```diff
@@ -101,6 +101,7 @@ export default class App extends Component<Props> {
           onExportAnnotationCommand={this.onExportAnnotationCommand}
           onZoomChanged={this.onZoomChanged}
           readOnly={false}
+          enableReadingModeQuickMenu={true}
```
Comment (ama-pdftron) [inline example/App.js:104]: Please revert this line, it should be part of testing step

**Review by ama-pdftron — CHANGES_REQUESTED** [review]

**Comment by ama-pdftron** [comment]
image
This doesn't seem to be working

File: ios/RNTPTDocumentView.m [hunk 4]
```diff
@@ -6243,6 +6243,11 @@ - (void)openThumbnailsView
 
 #pragma mark - Hygen Generated Props/Methods
 
+- (void)setEnableReadingModeQuickMenu:(BOOL)enabled
```
Comment (ama-pdftron) [inline ios/RNTPTDocumentView.m:6246]: Are these iOS changes necessary for an Android only feature?

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java [hunk 5]
```diff
@@ -150,6 +150,8 @@ public class DocumentView extends com.pdftron.pdf.controls.DocumentView2 {
     private boolean mUseStylusAsPen = true;
     private boolean mSignWithStamps;
 
+    private boolean mEnableQuickMenu = true;
```
Comment (ama-pdftron) [inline android/src/main/java/com/pdftron/reactnative/views/DocumentView.java:153]: Not a great name for the variable, you need to mention reading/reflow mode.

**Review by ama-pdftron — CHANGES_REQUESTED** [review]

File: android/src/main/java/com/pdftron/reactnative/views/DocumentView.java [hunk 6]
```diff
@@ -3098,6 +3104,7 @@ public void onTabDocumentLoaded(String tag) {
 
         getToolManager().setStylusAsPen(mUseStylusAsPen);
         getToolManager().setSignSignatureFieldsWithStamps(mSignWithStamps);
+        getToolManager().setReflowTextSelectionMenuEnabled(mEnableQuickMenu);
```
Comment (ama-pdftron) [inline android/src/main/java/com/pdftron/reactnative/views/DocumentView.java:3107]: `mEnableQuickMenu` -> ` mEnableReadingModeQuickMenu`

**Review by ama-pdftron — CHANGES_REQUESTED** [review]

**Review by ama-pdftron — APPROVED** [review]

**Pull request merged**